### PR TITLE
 multiple select for dropdown, ngx-toastr for error handling, fixed bug with selecting teamlead

### PR DESCRIPTION
**Pull request by olenakachmar**

### CODE REVIEW

File: client/src/app/common/models/update-user.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ export class UpdateUser {
     this.phone = user.phone;
     this.type = user.type;
     this.manager = user.manager._id;
-    this.teamlead = user.teamlead ? user.teamlead._id : undefined;
+    if (user.teamlead) {this.teamlead = user.teamlead._id;}
```
Comment (idyac) [inline client/src/app/common/models/update-user.ts]: format

File: client/src/app/page/create-update-user-page/create-update-side-bar-info/create-update-side-bar-info.component.ts [hunk 2]
```diff
@@ -64,18 +62,18 @@ export class CreateUpdateSideBarInfoComponent implements OnInit, OnDestroy {
...
     this.ifChosenHrDepartment = id === '5cb9c437b5cfd134acc5783e';
     this.positions = this.departments
-      .filter(elem => elem._id === id)[0].position
+      .find(elem => elem._id === id).position
       .map(e => new OptionPair(e, e));
     this.userService.chosenDepartment.emit();
   }
+
+  ngOnDestroy(): void {
+    this.destroy$.next();
```
Comment (idyac) [inline client/src/app/page/create-update-user-page/create-update-side-bar-info/create-update-side-bar-info.component.ts]: do you really need next here?

File: client/src/app/page/create-update-user-page/create-update-side-bar-info/dropdown-info/dropdown-info.component.html [hunk 3]
```diff
@@ -1,6 +1,7 @@
-<div class="btn-group" dropdown placement="bottom right">
+<script src="dropdown-info.component.ts"></script>
+<div class="btn-group" dropdown placement="bottom right" [autoClose]="roles ? false : true">
   <button id="button-basic" dropdownToggle type="button" class="btn btn-primary dropdown-toggle option-el"
-          [ngClass]=" !required ? []: title === 'Choose' ? ['border-red'] : ['border-green']"
+          [ngClass]=" !required ? ['border-blue']: title === 'Choose' ? ['border-red'] : ['border-green']"
```
Comment (idyac) [inline client/src/app/page/create-update-user-page/create-update-side-bar-info/dropdown-info/dropdown-info.component.html]: hard to read. move this logic to js.

File: client/src/app/page/create-update-user-page/create-update-side-bar-info/dropdown-info/dropdown-info.component.ts [hunk 4]
```diff
@@ -24,7 +38,33 @@ export class DropdownInfoComponent implements OnInit, OnChanges {
       this.title = 'Choose';
     }
     if (this.positionIdentifier === 'position') {
-      this.userService.chosenDepartment.subscribe(this.title = 'Choose');
+      this.userService.chosenDepartment.subscribe(() => {
+        this.title = 'Choose';
+      });
+    }
+    if (this.updateRolesList) {
+      this.updateRolesList.map(elem => {
```
Comment (idyac) [inline client/src/app/page/create-update-user-page/create-update-side-bar-info/dropdown-info/dropdown-info.component.ts]: forEach

File: client/src/app/page/create-update-user-page/create-update-side-bar-info/dropdown-info/dropdown-info.component.ts [hunk 5]
```diff
@@ -24,7 +38,33 @@ export class DropdownInfoComponent implements OnInit, OnChanges {
...
+        elem === 'manager' ? this.selectManagerRole = true : this.selectTeamleadRole = true;
```
Comment (idyac) [inline client/src/app/page/create-update-user-page/create-update-side-bar-info/dropdown-info/dropdown-info.component.ts]: move this logic to getSelectManagerRole or something else

File: client/src/app/page/create-update-user-page/create-update-side-bar-info/create-update-side-bar-info.component.ts (hunk 2 above)
Comment (olenakachmar) [reply client/src/app/page/create-update-user-page/create-update-side-bar-info/create-update-side-bar-info.component.ts]: this unsubscribes from the observable, isn't it?

**Pull request merged**